### ISSUE TITLE
Incorrect CA1033 message for interfaces

### ISSUE THREAD

**Issue by znakeeye** [dotnet/roslyn-analyzers#7582]
Using `VS 17.11.4`, targeting `.NET 8`.

The following code produces `CA1033: Make 'IFoo' sealed`. However, you cannot make an interface sealed. Clearly, this warning is either a false positive or has the wrong message.

```c#
namespace MyLib;

public interface IBar
{
    string Baz { get; }
}

public interface IFoo : IBar
{
    string IBar.Baz => "";
}
```

![Image](https://github.com/user-attachments/assets/cdf76356-8ec5-4cdf-8ca6-c039407b7aed)

To enable the warning, add this to the project file:
```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <LangVersion>latest</LangVersion>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
    
  </PropertyGroup>

</Project>
```
